# Maddalena's Notebook
## Open Science 2022/2023

## 23/03/23 

I started exploring the course material and the data sources we will need to use according to our research question

<b>Research Question</b>
* What is the coverage of publications in Social Science and Humanities (SSH) journals 
(according to ERIH-PLUS) included in OpenCitations Meta? 
* What are the disciplines that have more publications? 
* What are countries providing the largest number of publications and journals? 
* How many of the SSH journals are available in Open Access according to the data in DOAJ?

<b>data sources</b>
* <a href= "http://opencitations.net/meta">OpenCitations Meta</a>
* <a href= "https://kanalregister.hkdir.no/publiseringskanaler/erihplus/">ERIH-PLUS</a>
* <a href= "https://doaj.org/">DOAJ</a>

One of the first problems to address in our research will be how to retrieve the data, whether through REST API or by a dump download, hence I tried to figure out how a REST API works and then I tried making some requests to OpenCitations REST API.
Since we will need to access a large amount of data but at the moment I see no way of doing it manually, I am wondering if there are way of doing it by accessing the REST API with Python and I plan to look into it.


I downloaded ERIH-PLUS list of approved journals as a dump and I tried to explore the dataset

In [12]:
import pandas as pd

approvedSSH = pd.read_csv("F:\DATAvarie\ERIHPLUSapprovedJournals.csv", encoding = "utf-8", sep=";")
print(len(approvedSSH))
approvedSSH.info()
approvedSSH

11065
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11065 entries, 0 to 11064
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Journal ID              11065 non-null  int64 
 1   Print ISSN              8741 non-null   object
 2   Online ISSN             9510 non-null   object
 3   Original Title          11065 non-null  object
 4   International Title     11065 non-null  object
 5   Country of Publication  10945 non-null  object
 6   ERIH PLUS Disciplines   11065 non-null  object
 7   OECD Classifications    11065 non-null  object
 8   [Last Updated]          11065 non-null  object
dtypes: int64(1), object(8)
memory usage: 778.1+ KB


Journal ID Print ISSN Online ISSN  \
0          486254  1989-3477         NaN   
1          488561        NaN   2341-0515   
2          504135        NaN   2068-3472   
3          495209  2250-4591   2346-9986   
4          488332        NaN   0719-5737   
...           ...        ...         ...   
11060      486933  1995-4328   1995-4336   
11061      496777  2082-4939         NaN   
11062      502441  1409-939X   1857-968X   
11063      496467        NaN   2535-0390   
11064      491147  2309-1606         NaN   

                                          Original Title  \
0                     @tic.revista d'innovació educativa   
1      [i2] Investigación e Innovación en Arquitectur...   
2                                        [Inter]sections   
3                 +E: Revista de Extensión Universitaria   
4                                                 100-Cs   
...                                                  ...   
11060  ΣΧΟΛΗ (Schole). Philosophskoe antikovedenie i ...   
11061              Wrocławsko-Lwowskie Zeszyty Prawnicze   
11062                 ЕтноАнтропоЗум / EthnoAnthropoZoom   
11063                              Русистика без граници   
11064          Філософія освіти. Philosophy of Education   

                                     International Title  \
0                     @tic.revista d'innovació educativa   
1      [i2] Investigación e Innovación en Arquitectur...   
2                                        [Inter]sections   
3                 +E: Revista de Extensión Universitaria   
4                                                 100-Cs   
...                                                  ...   
11060  ΣΧΟΛΗ (Schole). Ancient philosophy and the cla...   
11061           Вроцлавсько-львівський юридичний збірник   
11062                 ЕтноАнтропоЗум / EthnoAnthropoZoom   
11063  Русистика без границ; Russian studies without ...   
11064          Філософія освіти. Philosophy of Education   

      Country of Publication  \
0                      Spain   
1                      Spain   
2                    Romania   
3                  Argentina   
4                      Chile   
...                      ...   
11060                 Russia   
11061                 Poland   
11062              Macedonia   
11063               Bulgaria   
11064                Ukraine   

                                   ERIH PLUS Disciplines  \
0      Interdisciplinary research in the Social Scien...   
1      Art and Art History, Cultural Studies, Human G...   
2      Gender Studies, Cultural Studies, Literature, ...   
3      Interdisciplinary research in the Humanities, ...   
4      Interdisciplinary research in the Social Sciences   
...                                                  ...   
11060  Archaeology, Art and Art History, Classical St...   
11061  History, Business and Management, Economics, L...   
11062  Anthropology, Cultural Studies, Religious Stud...   
11063  History, Cultural Studies, Linguistics, Litera...   
11064     Pedagogical & Educational Research, Philosophy   

                                    OECD Classifications       [Last Updated]  
0            Educational Sciences; Other Social Sciences  2015-06-25 13:48:26  
1      Arts (Arts, History of Arts, Performing Arts, ...  2016-04-18 17:34:55  
2      Languages and Literature; Other Humanities; So...  2022-10-18 08:40:28  
3                Other Humanities; Other Social Sciences  2023-02-02 17:14:12  
4                                  Other Social Sciences  2018-04-26 16:33:26  
...                                                  ...                  ...  
11060  Arts (Arts, History of Arts, Performing Arts, ...  2016-02-09 09:58:08  
11061  Economics and Business; History and Archaeolog...  2020-09-28 15:44:29  
11062  Languages and Literature; Philosophy, Ethics a...  2022-04-07 09:17:03  
11063  Educational Sciences; History and Archaeology;...  2019-04-03 10:04:29  
11064  Educational Sciences; Philosophy, Ethics an

Nonetheless these informations are about journals and not about publication. I am thinking we will need a list of DOI of the publications indexed in ERIH-PLUS, I have explored a bit but I can't seem to find articles' downloadable data. This will be something to address in the first group meeting.


## 27/03/23 

I read Emerald publishing guidelines on how to write an abstract. 

According to the key points that an abstract should feature, I revised the sources provided by the teacher on our research question subject and did some further research to better focus what is the purpose and value of our research.

In particular, since the domain of our research question is Scientometrics and inquires about the coverage of SSH journal in OpenCitatins and how many of those are open access, I noted down some articles I skimmed and plan to read in the future:

- <a href="https://www.nature.com/articles/s41599-018-0149-x">The possibility and desirability of replication in the humanities</a>
- <a href="https://scholarlypublications.universiteitleiden.nl/handle/1887/65315">Exploration of reproducibility issues in scientometric research</a>
- <a href="https://direct.mit.edu/qss/article/3/4/953/113634/A-quantitative-and-qualitative-open-citation">A quantitative and qualitative open citation analysis of retracted articles in the humanities</a>
- <a href="https://www.degruyter.com/document/doi/10.1515/opis-2018-0013/html">Scholarly Communication Practices in Humanities and Social Sciences: A Study of Researchers' Attitudes and Awareness of Open Access</a>

Finally, I started reasoning on the first point of the abstract "Purpose" by formulating some on point questions to share with the group later today, as well as my personal answer, to direct us for writing our abstract. 

* what is the problem we need to solve? why is it relevant? What is the main finding?
<center>The problem to solve is to understand the state of the art of Social Science and Humanities journals citation data presence in OpenCitation databases. <br> It is relevant because of citations centrality in ssh domain and the possibility of conducting further research on available data on citations. How does Open access relate to that?<br>
The main finding will be 1.providing an up do date overview on citation data and oppennes in the ssh domain. 2. creating a software that can make these data available and keep them up to date for enhancing further research on the topic</center>

<hr>

### slef-notes
* How many of the Open Access journals adopt Article processing charge (= make people pay a fee to publish open access?). What is that for? Where else could the open access system sustain itself without having authors pay to publish?

<b>Notes from <a href="https://www.emeraldgrouppublishing.com/how-to/authoring-editing-reviewing/write-article-abstract">Emerald group publishing guidelines</a></b>

maximum of 250 words

<b>Points to feature</b>
* <b>Purpose</b>: This is where you explain why you undertook this study. explain the problem that you have solved.  let readers know why you chose to study this topic or problem and its relevance. Let them know what your key argument or main finding is.
* <b>Study design, methodology, approach</b>: Let readers know exactly what you did to reach your results.  Used tools, methods, protocols or datasets
* <b>Findings</b>: what you found during your study, whether it answers the problem you set out to explore, and whether your hypothesis was confirmed. Clear, give exact figures and not generalizations.
* <b>Originality, value</b>: analysis of the value of your results. Ask colleagues whether your analysis is balanced and fair. Conjecture what future research steps could be. 

<b>Include</b>
* Reasearch limitations/implications
* Practical implications
* Social implications


After the group meeting - where we shared our current understanding of the research to carry and also envisioned possible study designs and various technical approaches - I offered to write a coherent first draft of the abstract considering the material and perspectives we shared. This first draft is pushed in the repository for my teammates to read and rework on, so we can agree on a final version to present on Wednesday.


## Abstract first draft
<span style="color: red">currently around 400 words, to cut</span>

* <b>Purpose</b>: In the domain of Social Science and Humanities, citations hold a central role in measuring reliability and interconnectedness of a publication as well as its long term impact. Referencing works that inspired a particular work provide the reader with the possibility of accessing primary sources, hence, enhancing scholarly communication and community evaluation of a research.
This project aims to assess the state of the art of Social Science and Humanities journals citation data coverage in OpenCitation databases and investigate how many of these are also Open Access. The research provides further insight into the available data, evaluating which countries and disciplines provide the larger number of publications and journals.

* <b>Study design and methodology</b>: We use data from multiple sources: OpenCitations Meta, a database that stores and delivers bibliographic metadata about citations; ERIH-PLUS, an academic Social Science and humanities journal index; DOAJ, the directory of open access journals to retrieve data about open access journals. These data are fetched, filtered, and processed by means of Python Programming Language. The code will be published and available in a GitHub repository, along with instructions on how to use it, to provide users full access to reproduce this analysis at any moment in time. Finally, output data will be visualized in an intuitive and user friendly way.

* <b>Findings</b>: Considering the diverse problematics and the reluctance that revolve around opening the scholarly knowledge in the SSH domain, we expect to find an increasing but still not enough full open access journals. Nonetheless, we expect this deficiency to be in part counterbalanced by a significant number of open citation metadata of SSH publications, provided by OpenCitations  database. <span style="color: red">to inquire</span>

* <b>Originality</b>: Our study provides valuable insights into the current state of the openness of Social Science and Humanities publications and provide reusable data for enhancing further research about Open Science in the Scientometrics domain.
Furthermore, we believe that increased open access to SSH research can enable broader societal engagement, foster cross-disciplinary collaboration, and promote the dissemination of knowledge to a wider audience.

* <b>Limitations</b>: The research is based on the current state and reliability of publicly available data, which might not encompass the entire landscape of SSH publications. These data may and will probably change over time or might come with errors we overlook, despite our best efforts to be as precise an meticulous as possible.

Keywords: Open Access, Social Science and Humanities, OpenCitations Meta, ERIH-PLUS, DOAJ, scholarly communication, open science, citation

## 02/04/23 

Today I revised 29/03 lesson about FAIR and open data and summarized/organized my handwritten notes + slides in a unique txt document.
While revising I noted down some questions and doubts to share and discuss with the group during our next meeting, where we will start working on our Data Management Plan.
<hr>
<b>Unclear issues</b>

- What are practical examples of vocabularies that use FAIR principles? (Interoperability principle 2)
- (Meta)data include qualified references to other (meta)data (?)
About the DMP
- What data (beside csv datasets) will our research produce?
- Do we need to add metadata to our data?
- what metadata should a software have, are there standards?
- Datasets to describe in our DMP are source datasets or our final processed dataset?

## 04/04/23 

In order to fill up the sections of the DMP I am in charge of, I first read a <a hred="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004525">Ten simple rules for creating a good data management PLAN</a> and wrote a small summary, highlighting the parts that may concern us the most.

Then I started writing the DMP first dataset description.
1. data summary 
2. reused data
3. fair data

During the compilation of the Dataset description I stopped at some questions, since I realized I was not entirely sure how to answer them:

<b>1.1.3 What are the formats of the described generated/collected data?</b><br>
I am not sure if a dataset description should be specific for one kind of data or if it can include all kind of data produced during the research? I think --> our generated data will be diffused in different ways, a dataset in the classical term for sure, but then also probably a textual description analysis and likely some visualiztions (maybe a platform? A website? a pdf?)

I looked into some other resources to help me better understand the specific of different data formats
* <a href="https://ukdataservice.ac.uk/learning-hub/research-data-management/format-your-data/recommended-formats/">File formats recommended by the UK Data Service</a>
* <a href="http://www.docs.is.ed.ac.uk/docs/data-library/EUDL_RDM_Handbook.pdf">Edinburgh University Data Library Research Data Management Handbook</a>
* <a href="https://dmptool-stg.cdlib.org/general_guidance">DMPTool</a>

In the dropdown menu I chose "Discipline specific formats" since it seemed to me the most suitable to our type of data. All the formats we will use are in common usage by the research community, even though not "discipline specific" in a narrow sense.

<b>2.1.1 Are you re-using the described data and how?</b><br>
since we are using public structured data I initially though this meant we were reusing data. Anyway, on a second thought, since they seem to intend "reused data" as data that are outcomes of other researches, this is not true for us: our source data is downloaded from public directories, aggregators and databases.

<b>3.1.1.1 Will you use metadata to describe the data?</b>
I am familiar with DCAT standard but I though it could be worth looking into other standards amongst the ones present in the dropdown menu. I had a quick look at <a href="https://www.w3.org/TR/vocab-data-cube/#outline">The RDF data cube vocabulary</a> and at <a href="https://schema.datacite.org/meta/kernel-4.4/doc/DataCite-MetadataKernel_v4.4.pdf">The DataCite metadata schema</a>.
Reading the RDF data cube vocabulary, made me think about whether we will be also producing statystical data and in what format they need to be published (I think csv is fine, but also could be visualizations) and with which standard they should be described (DCAT is also fine but we could look into SDMX)<br>
<b>N.B.</b> DCAT has the class dcat:Resource which is a general catalogue resource: "It is strongly recommended to use a more specific sub-class. When describing a resource which is not a dcat:Dataset or dcat:DataService, it is recommended to create a suitable sub-class of dcat:Resource, or use dcat:Resource with the dct:type property to indicate the specific type." this would allow as to include non-datasets resurces as data in our catalogue too.

<b>3.1.1.12 Will you provide searchable metadata for the described data?</b>
How can we make our metadata searchable?

As it turns out, many other answers require further group discussion. So far, considering these questions led me to note down useful resources to envision possible answers.

<hr>

* <a href="https://www.openaire.eu/find-trustworthy-data-repository">data repositories</a>
* <a href="http://www.openarchives.org/OAI/2.0/guidelines.htm">OAIP-MH</a>





## 05/04/23 

I filled up the FAIR section of the software dataset in our DMP. For applying FAIR principles to software data, this resource was very helpful: <a href="https://fair-software.nl/">Five reccommendation for fair software</a>

Later, Sebastiano and I had a meeting to revise together every part of the DMP and discuss the parts we individually filled.

## 07/04/23 

I had a look at Ali's additions in the DMP than <a href="https://zenodo.org/record/7809055#.ZDBTZ3ZBxEY">uploaded it</a> on Zenodo 
The first version of our DMP has the following doi:10.5281/zenodo.7809055 

Then, I revised 30/03 lesson about FAIR and open data and summarized/organized my handwritten notes + slides in a unique txt document.

This is a draft of a methodology for a preliminary Literature Review (Introduction to a primary research topic) for our research.
Criteria: 
* key words: OoenCitations, Open Access, SSH Journals
* not before year 2020
* I will read the selected paper's abstract and check how detailed is their methodology section (if present) or if they are related to an independent methodology publications. Then I will select among those, three papers to read more thoroughly.
* Studies providing an independent methodology publication will be preferred.

Finally, I exploerd a bit CWL and protocols.io, running <a href="https://www.protocols.io/view/a-methodology-for-gathering-and-annotating-the-raw-5qpvo5pmbl4o/v2">this</a> workflow + downloaded OpenCitations Meta data dump


## 08/04/23 
I did some further exploration of our datasets in order to better understand how to structure our workflow and share possible ideas or problems during this evening's group meeting.

In [13]:
# exploring ERIH PLUS disciplines for answering research question:
# what are the disciplines that have more publications ?
allcombination = set(approvedSSH['ERIH PLUS Disciplines'])
allcombination = list(allcombination)
disciplines = set()
for el in allcombination:
    if ',' in el:
        disc = (el.split(', '))
        for d in disc:
            disciplines.add(d)
    else: 
        disciplines.add(el)
print(len(disciplines)) #30
print(disciplines)

#OECD classification is another disciplinary classification so probably we will need to map those
allOECDcombination = set(approvedSSH['OECD Classifications'])
allOECDcombination = list(allOECDcombination)
disciplinesOECD = set()
for e in allOECDcombination:
    if ';' in e:
        dis = (e.split('; '))
        for d in dis:
            disciplinesOECD.add(d)
    else: 
        disciplinesOECD.add(e)
print(len(disciplinesOECD)) #14
print(disciplinesOECD)



30
{'History & Philosophy of Science', 'Psychology', 'Anthropology', 'Social Statistics and Informatics', 'Art and Art History', 'Linguistics', 'Business and Management', 'Interdisciplinary research in the Social Sciences', 'Archaeology', 'Philosophy', 'History', 'Gender Studies', 'Sociology', 'Demography', 'Pedagogical & Educational Research', 'Law', 'Human Geography and Urban Studies', 'Science and Technology Studies', 'Religious Studies and Theology', 'Musicology', 'Literature', 'Library and Information Science', 'Cultural Studies', 'Political Sciences and International Relations', 'Interdisciplinary research in the Humanities', 'Economics', 'Environmental Studies', 'Media Studies and Communication', 'Film and Theatre Studies', 'Classical Studies'}
14
{'Social and Economic Geography', 'Media and Communications', 'Languages and Literature', 'Law', 'Psychology', 'Economics and Business', 'Other Humanities', 'Educational Sciences', 'Philosophy, Ethics and Religion', 'History and Archae

<a href="http://help.prod-incites.com/inCites2Live/filterValuesGroup/researchAreaSchema/oecdCategoryScheme/oecd.html">OECD classification</a> is another disciplines classification

As far as ISSN/ESBN are concerned, some of them might have null values as we've seen:

0.   Journal ID : 11065 non-null  int64 
1.   Print ISSN : 8741 non-null   object
2.   Online ISSN : 9510 non-null   object

same goes for Countries:

3. Country of Publication : 10945 non-null (out of 11065 journals)

We need to make sure that at least on of the field "Print ISSN" / "Online ISSN" is non-null because this is the unique identifier that will allow us to connect with OpenCitation meta dataset

In [14]:
#let's check if all 11065 journals are unique through ERIH PLUS unique ID
journals = set(approvedSSH['Journal ID'])
len(journals) #11065
#check how many NaN values in ISSN
nanISSN = approvedSSH['Print ISSN'].isna().sum()
print(nanISSN)
nanISSNOnline = approvedSSH['Online ISSN'].isna().sum()
print(nanISSNOnline)
#are there cases when both of them are null?
noISSN = 0
for idx, row in approvedSSH.iterrows():
    if pd.isna(row['Print ISSN']) and pd.isna(row['Online ISSN']):
        noISSN += 1
print(noISSN) #0 
#pare di no

#also check how many countries are na
noCountry = approvedSSH['Country of Publication'].isna().sum()
print(noCountry) #oioioi 120 


2324
1555
0
120


missing country values might be a problem to answer our research question "What are countries providing the largest number of publications and journals?" but we might be able to retreive them for other datasets.

Now let's explore DOAJ

In [15]:
doaj = pd.read_csv("F:\DATAvarie\journalcsv__doaj_20230404_1935_utf8.csv", encoding="UTF-8") 
doaj.head()
doaj.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19172 entries, 0 to 19171
Data columns (total 54 columns):
 #   Column                                                                       Non-Null Count  Dtype  
---  ------                                                                       --------------  -----  
 0   Journal title                                                                19172 non-null  object 
 1   Journal URL                                                                  19172 non-null  object 
 2   URL in DOAJ                                                                  19172 non-null  object 
 3   When did the journal start to publish all content using an open license?     19171 non-null  float64
 4   Alternative title                                                            7458 non-null   object 
 5   Journal ISSN (print version)                                                 11088 non-null  object 
 6   Journal EISSN (online version)        

In [16]:
#let's select columns that interest 
doaj = doaj[["Journal title", "Journal ISSN (print version)", "Journal EISSN (online version)", "Keywords", "Publisher", "Country of publisher", "Persistent article identifiers", "Does the journal comply to DOAJ's definition of open access?", "Subjects"]]
doaj.head()

Journal title Journal ISSN (print version)  \
0  Anais da Academia Brasileira de Ciências                    0001-3765   
1                                      ACME                    0001-494X   
2                  Acta Biochimica Polonica                    0001-527X   
3                Acta Dermato-Venereologica                    0001-5555   
4                 Acta Médica Costarricense                    0001-6012   

  Journal EISSN (online version)  \
0                      1678-2690   
1                      2282-0035   
2                      1734-154X   
3                      1651-2057   
4                      2215-5856   

                                            Keywords  \
0  biological sciences, exact and earth sciences,...   
1  italian literature, classic literature, lingui...   
2  molecular biology, biophysics, bioinformatics,...   
3  sexually transmitted infections, psoriasis, ps...   
4  medicine, public health, medical sciences, health   

                                      Publisher Country of publisher  \
0               Academia Brasileira de Ciências               Brazil   
1              Università degli Studi di Milano                Italy   
2                    Polish Biochemical Society               Poland   
3                       Medical Journals Sweden               Sweden   
4  Colegio de Médicos y Cirujanos de Costa Rica           Costa Rica   

  Persistent article identifiers  \
0                            DOI   
1                       DOI, NBN   
2                            DOI   
3                            DOI   
4                            NaN   

  Does the journal comply to DOAJ's definition of open access?  \
0                                                Yes             
1                                                Yes             
2                                                Yes             
3                                                Yes             
4                                                Yes             

                                            Subjects  
0                                            Science  
1                                      General Works  
2  Science: Chemistry: Organic chemistry: Biochem...  
3                              Medicine: Dermatology  
4                                           Medicine

Let's try to merge ERIH-PLUS and DOAJ data on the ISSN, first in print, then online, to keep only SSH Journals that are also in DOAJ directory

In [17]:
oa_sshPrint = pd.merge(doaj, approvedSSH, left_on="Journal ISSN (print version)", right_on="Print ISSN", how="inner")
oa_sshPrint.head() #18789539 which seems a lot, what is repeating?
# we can observe that "country of publisher" and "country of publication" have the same values
oa_sshOL = pd.merge(doaj, approvedSSH, left_on="Journal EISSN (online version)", right_on="Online ISSN", how="inner")
oa_sshOL.head() #1501144 

Journal title Journal ISSN (print version)  \
0                   Acta Psychologica                    0001-6918   
1                           Afro-Ásia                    0002-0591   
2  Anali Pravnog Fakulteta u Beogradu                    0003-2565   
3                      Análise Social                    0003-2573   
4                  Angle Orthodontist                    0003-3219   

  Journal EISSN (online version)  \
0                      1873-6297   
1                      1981-1411   
2                      2406-2693   
3                      2182-2999   
4                            NaN   

                                            Keywords  \
0  clinical psychology, health psychology, cognit...   
1  african studies, diaspora studies, race relati...   
2  international law, law, civil law, criminal la...   
3  sociology, social conditions, political scienc...   
4                odontology, dentistry, orthodontics   

                                           Publisher Country of publisher  \
0                                           Elsevier          Netherlands   
1  Universidade Federal da Bahia - Centro de Estu...               Brazil   
2  University of Belgrade, Faculty of Law, Belgra...               Serbia   
3  Instituto de Ciências Sociais da Universidade ...             Portugal   
4        E.H Angle Education and Research Foundation        United States   

  Persistent article identifiers  \
0                            DOI   
1                            DOI   
2                            DOI   
3                            NaN   
4                            DOI   

  Does the journal comply to DOAJ's definition of open access?  \
0                                                Yes             
1                                                Yes             
2                                                Yes             
3                                                Yes             
4                                                Yes             

                                            Subjects  Journal ID Print ISSN  \
0       Philosophy. Psychology. Religion: Psychology      340169  0001-6918   
1  History (General) and history of Europe: Histo...      503556  0002-0591   
2                                                Law      501706  0003-2565   
3  Political science | Social Sciences: Sociology...      487446  0003-2573   
4                                Medicine: Dentistry      486254  1989-3477   

  Online ISSN                      Original Title  \
0   1873-6297                   Acta Psychologica   
1   1981-1411                           Afro-Ásia   
2   2406-2693  Anali Pravnog fakulteta u Beogradu   
3   2182-2999                      Análise Social   
4         NaN  @tic.revista d'innovació educativa   

                        International Title Country of Publication  \
0                         Acta Psychologica            Netherlands   
1                                 Afro-Asia                 Brazil   
2  Annals of the Faculty of Law in Belgrade                 Serbia   
3                            Análise Social               Portugal   
4        @tic.revista d'innovació educativa                  Spain   

                               ERIH PLUS Disciplines  \
0                                         Psychology   
1  Anthropology, History, Cultural Studies, Socio...   
2  History, History & Philosophy of Science, Econ...   
3  Sociology, Anthropology, History, Political Sc...   
4  Interdisciplinary research in the Social Scien...   

                                OECD Classifications       [Last Updated]  
0                                         Psychology  2023-02-02 17:14:12  
1  History and Archaeology; Languages and Literat...  2022-04-24 14:23:07  
2  Economics and Business; History and Archaeolog...  2021-05-19 11:21:25  
3  History and Archaeology; Political Science; So...  2016-04-10 12:16:35  
4        Educational Sciences; Other Soc

We will probably need to reorganize journal identifiers to actually get all possible matches<br>
Now check if ISSN are repeated in our two dataframes:

In [18]:
uniqueISSN = set(approvedSSH['Print ISSN'])
print(len(uniqueISSN)) #8742

uniqueISSN1 = set(doaj['Journal ISSN (print version)'])
print(len(uniqueISSN1)) #11089

#they both have 8742 unique ISSN sounds wierd
approvedSSH.describe(include="all")
doaj.describe(include="all")

8742
11089


Journal title Journal ISSN (print version)  \
count                   19172                        11088   
unique                  19037                        11088   
top     Revista de Psicología                    0001-3765   
freq                        5                            1   

       Journal EISSN (online version)   Keywords Publisher  \
count                           18209      19172     19171   
unique                          18209      18077      8719   
top                         1678-2690  education  Elsevier   
freq                                1         49       643   

       Country of publisher Persistent article identifiers  \
count                 19172                          14359   
unique                  131                             45   
top               Indonesia                            DOI   
freq                   2235                          14081   

       Does the journal comply to DOAJ's definition of open access?  Subjects  
count                                               19157               19172  
unique                                                  1                3695  
top                                                   Yes            Medicine  
freq                                                19157                 515

I run out of memory so I explored OpenCitation Meta dump manually
<table>
<tr>
<th>id</th>
<th>title</th>
<th>author</th>
<th>issue</th>
<th>volume</th>
<th>venue</th>
<th>page</th>
<th>pub_date</th>
<th>type</th>
<th>publisher</th>
<th>editor</th>
</tr>
<tr>
<td>meta:br/060209 doi:10.4230/lipics.approx/random.2020.19</td>
<td>Distributed Testing Of Graph Isomorphism In The CONGEST Model</td>
<td>Levi, Reut [meta:ra/0610110096 orcid:0000-0003-3167-1766]; Medina, Moti [meta:ra/0612046435 orcid:0000-0002-5572-3754]</td>
<td></td>
<td></td>
<td>[meta:br/060182 issn:1868-8969]</td>
<td></td>
<td>2020</td>
<td>report</td>
<td>Schloss Dagstuhl - Leibniz-Zentrum Für Informatik [meta:ra/0605251]</td>
<td>Byrka, Jarosław [meta:ra/069044096 orcid:0000-0002-3387-0913]; Raghu Meka [meta:ra/0605252]</td>
</tr>
</table>


As we can see, meta has its own internal identifiers (<a href="https://opencitations.net/meta">OMID</a>), <q>The entities subject to deduplication and associated with an OMID are identifiers (abbr. id), agent roles (i.e., authors, editors, publishers, abbr. ar), responsible agents (i.e., people and organisations, abbr. ra), resource embodiments (i.e., pages, abbr. re), venues, volumes, and issues (which are all bibliographic resources, abbr. br).</q><br>
OMIDS areseparated from ISSN and other identifiers by a blank space.<br>

Later, we had the group meeting where we started compiling a workflow in the protocols.io platform.

## 12/04/23 

I completed a first draft of the workflow adding the steps that were missing after our group meeting. I left the vizualization part to be completed by my teammates, as well as any other changes or feedback they may have on the workflow.

This first draft is saved in a pdf: https://drive.google.com/file/d/1MSGXQWTWEzxym7zgHx1CcS2-nI60Nux-/view?usp=sharing 
